aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -637,10 +637,16 @@ void THTensor_(arange)(THTensor *r_, accreal xmin, accreal xmax, accreal step) {
   scalar_t i = 0;
 
   THArgCheck(step > 0 || step < 0, 3, "step must be nonzero");
+  THArgCheck(std::isfinite(static_cast<double>(xmin)) &&
+              std::isfinite(static_cast<double>(xmax))
+              , 1, "unsupported range: ", xmin, " -> ", xmax);
```

Comment:
I checked the bug report, and the PR comments, but I don't see why you need to cast the quantity to a double before checking if it's finite. I'm pretty sure `std::isfinite` has overloads for the integral types too. Is there something I'm missing?

Reply:
hmm...I thought I remembered getting an "ambiguous call to overloaded function" when I didn't cast, but, not seeing that now.

Reply:
ah, yeah, looks like that 'ambiguous' error only occurs on windows. I'll need to re-add the cast:

```
15:26:10 C:\Program Files (x86)\Windows Kits\10\include\10.0.16299.0\ucrt\corecrt_math.h(402): error C2668: 'fpclassify': ambiguous call to overloaded function
15:26:10 C:\Program Files (x86)\Windows Kits\10\include\10.0.16299.0\ucrt\corecrt_math.h(299): note: could be 'int fpclassify(long double) throw()'
15:26:10 C:\Program Files (x86)\Windows Kits\10\include\10.0.16299.0\ucrt\corecrt_math.h(294): note: or       'int fpclassify(double) throw()'
15:26:10 C:\Program Files (x86)\Windows Kits\10\include\10.0.16299.0\ucrt\corecrt_math.h(289): note: or       'int fpclassify(float) throw()'
15:26:10 C:\Program Files (x86)\Windows Kits\10\include\10.0.16299.0\ucrt\corecrt_math.h(402): note: while trying to match the argument list '(int64_t)'
15:26:10 c:\jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-build\aten\src\th\generic/THTensorMoreMath.cpp(642): note: see reference to function template instantiation 'bool isfinite<int64_t>(_Ty) throw()' being compiled
15:26:10         with
15:26:10         [
15:26:10             _Ty=int64_t
15:26:10         ]

```